stumpy/stumpi.py

```diff
@@ -125,26 +142,54 @@ def __init__(self, T, m, egress=True, normalize=True, p=2.0, k=1):
             The number of top `k` smallest distances used to construct the matrix
             profile. Note that this will increase the total computational time and
             memory usage when `k > 1`.
+
+        T_subseq_isconstant_func : function, default None
+            A custom, user-defined function that returns a boolean array that indicates
+            whether a subsequence in `T` is constant (True). The function must only take
+            two arguments, `a`, a 1-D array, and `w`, the window size, while additional
+            arguments may be specified by currying the user-defined function using
+            `functools.partial`. Any subsequence with at least one np.nan/np.inf will
+            automatically have its corresponding value set to False in this boolean
+            array.
         """
         self._T = core._preprocess(T)
         core.check_window_size(m, max_size=self._T.shape[-1])
         self._m = m
         self._k = k
 
+        if T_subseq_isconstant_func is None:
+            T_subseq_isconstant_func = core._rolling_isconstant
+        if not callable(T_subseq_isconstant_func):  # pragma: no cover
+            msg = (
+                "`T_subseq_isconstant_func` was expected to be a callable function "
+                + f"but {type(T_subseq_isconstant_func)} was found."
+            )
+            raise ValueError(msg)
+        self._T_subseq_isconstant_func = T_subseq_isconstant_func
+
         self._n = self._T.shape[0]
         self._excl_zone = int(np.ceil(self._m / config.STUMPY_EXCL_ZONE_DENOM))
         self._T_isfinite = np.isfinite(self._T)
         self._egress = egress
 
-        mp = stump(self._T, self._m, k=self._k)
+        self._T_subseq_isconstant = core.rolling_isconstant(
+            self._T, self._m, self._T_subseq_isconstant_func
+        )
+
+        mp = stump(
+            T_A=self._T,
```

Comment:
`T_A` and `m` are required arguments and so we do not need to specify the `T_A=` or `m=`. While it is nice to be explicit, I think it is pretty consistent throughout the codebase that `T_A` and `m` are required arguments and may be omitted. It's a little bit different if we are calling a function that is rarely used and we need a reminder what the first two arguments are but this is not one of those cases.

Reply:
> While it is nice to be explicit, I think it is pretty consistent throughout the codebase that `T_A` and `m` are required arguments and may be omitted. It's a little bit different if we are calling a function that is rarely used and we need a reminder what the first two arguments are but this is not one of those cases.

I agree. The reason that I tried to be explicit here is that we see `T_A_subseq_isconstant=self._T_subseq_isconstant` when we call function `stump`, and I thought it might confuse reader as where that `T_A` comes from.

> it is pretty consistent throughout the codebase that T_A and m are required arguments

Now that you mentioned it, I think the developer probably knows that the first two arguments are `T_A` and  `m`. So, it should be clear. 


I will revise this part.